p2p/src/net/mock/backend.rs

```diff
@@ -133,14 +140,15 @@ impl Backend {
     /// `pending` to `peers` and the front-end is notified about the peer.
     async fn create_peer(
         &mut self,
-        socket: TcpStream,
+        socket: T::Socket,
         local_peer_id: types::MockPeerId,
         remote_peer_id: types::MockPeerId,
         role: peer::Role,
-        state: ConnectionState,
+        state: ConnectionState<T>,
     ) -> crate::Result<()> {
         let (tx, rx) = mpsc::channel(16);
-        let socket = socket::MockSocket::new(socket);
+        // TODO: FIXME:
+        //let socket = socket::MockSocket::new(socket);
```

Comment:
I think `socket` can be passed on `Peer` without converting it into a `MockSocket` first because `SocketService` provides everything that the `MockSocket` did